app/src/org/commcare/activities/DispatchActivity.java

```diff
@@ -314,7 +314,11 @@ private String getSessionEndpointAppId() {
     }
 
     private boolean getLaunchedFromConnect() {
-        return getIntent().getBooleanExtra(IS_LAUNCH_FROM_CONNECT, false);
+        boolean launchedFromConnect = getIntent().getBooleanExtra(IS_LAUNCH_FROM_CONNECT, false);
+        if(launchedFromConnect) {
```

Comment:
this is not needed as extras.remove doesn't do anything by itself if the key doesn;t exist.